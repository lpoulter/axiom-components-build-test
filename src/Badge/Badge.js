import PropTypes from 'prop-types';
import React, { Component } from 'react';
import classnames from 'classnames';
import Base from '../Base/Base';
import './Badge.css';

export default class Badge extends Component {
  static propTypes = {
    /** Content to appear inside the Badge */
    children: PropTypes.node,
    /** Colour of the Badge */
    color: PropTypes.oneOf([
      'faded',
      'highlight',
      'success',
      'error',
      'forbidden-planet',
      'tiny-clanger',
      'critical-mass',
      'fantastic-voyage',
      'paradise-lost',
      'serene-sea',
      'event-horizon',
      'electric-dreams',
      'outer-limits',
      'giant-leap',
      'moon-lagoon',
      'space-invader',
      'extraterrestrial',
      'terra-form',
      'primeval-soup',
      'future-shock',
      'sun-maker',
      'new-horizon',
      'blast-off',
      'crash-course',
      'solar-rust',
      'ground-control',
      'space-oddity',
      'rocky-planet',
      'deep-thought',
      'luna-dust',
    ]).isRequired,
    /**
     * Controls the full width appearance of the badge either all of the time,
     * with a value of `true` otherwise at one of the breakpoints specified.
     */
    full: PropTypes.oneOf([true, 'small', 'medium', 'large']),
    /** Click handler that applied clickable styling */
    onClick: PropTypes.func,
    /** The opacity of the background of the Badge */
    opacity: PropTypes.number,
  };

  static defaultProps = {
    color: 'faded',
    opacity: 1,
  };

  render() {
    const {
      children,
      color,
      full,
      onClick,
      opacity: o,
      ...rest
    } = this.props;

    const isFaded = color === 'faded';
    const textColor = isFaded || o < 1 ? undefined : 'night';
    const opacity = isFaded || o < 1 ? Math.max(.1, Math.min(isFaded ? .3 : .6, o)) : 1;
    const style = { opacity };
    const classes = classnames('ax-badge', {
      [`ax-badge--${color}`]: color,
      'ax-badge--clickable': onClick,
      'ax-badge--full': full === true,
      [`ax-badge--full--${full}`]: typeof full === 'string',
    });

    return (
      <Base { ...rest } Component="span" className={ classes } onClick={ onClick } textColor={ textColor }>
        <span className="ax-badge__background" style={ style } />
        <span className="ax-badge__content">{ children }</span>
      </Base>
    );
  }
}
