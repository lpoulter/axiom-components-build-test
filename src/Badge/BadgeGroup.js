import PropTypes from 'prop-types';
import React, { Component } from 'react';
import InlineGroup from '../InlineGroup/InlineGroup';

export default class BadgeGroup extends Component {
  static propTypes = {
    /** Badges inside the group */
    children: PropTypes.node,
  };

  render() {
    const { children, ...rest } = this.props;

    return (
      <InlineGroup space="x4" { ...rest } className="ax-badge-group">
        { children }
      </InlineGroup>
    );
  }
}
