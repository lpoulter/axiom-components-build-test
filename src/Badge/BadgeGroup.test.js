import React from 'react';
import renderer from 'react-test-renderer';
import Badge from './Badge';
import BadgeGroup from './BadgeGroup';

const getComponent = (props = {}) =>
  renderer.create(
    <BadgeGroup { ...props }>
      <Badge>Lorem ipsum</Badge>
    </BadgeGroup>
  );

describe('BadgeGroup', () => {
  it('renders with defaultProps', () => {
    const component = getComponent();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
