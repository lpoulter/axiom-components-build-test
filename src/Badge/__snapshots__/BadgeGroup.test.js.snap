// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BadgeGroup renders with defaultProps 1`] = `
<div
  className="ax-inline-group ax-badge-group ax-space--x4"
>
  <div
    className="ax-inline-group__spacer"
  >
    <span
      className="ax-badge ax-badge--faded"
    >
      <span
        className="ax-badge__background"
        style={
          Object {
            "opacity": 0.3,
          }
        }
      />
      <span
        className="ax-badge__content"
      >
        Lorem ipsum
      </span>
    </span>
  </div>
</div>
`;
